lib/auth.jsx

```diff
@@ -159,6 +178,118 @@ export class Auth extends Component {
     this.props.onAuthenticate(username, password);
   };
 
+  onWPLogin = () => {
+    const config = getConfig();
+    this.setupAuthWindow();
+
+    const redirectUrl = encodeURIComponent(config.wpcc_redirect_url);
+    this.authState = `app-${cryptoRandomString(20)}`;
+    const authUrl = `https://public-api.wordpress.com/oauth2/authorize?client_id=${
+      config.wpcc_client_id
+    }&redirect_uri=${redirectUrl}&response_type=code&scope=global&state=${
+      this.authState
+    }`;
+
+    this.authWindow.loadURL(authUrl);
+    this.authWindow.show();
+  };
+
+  setupAuthWindow = () => {
+    const remote = __non_webpack_require__('electron').remote; // eslint-disable-line no-undef
+    const BrowserWindow = remote.BrowserWindow;
+    const protocol = remote.protocol;
+    this.authWindow = new BrowserWindow({
+      width: 640,
+      height: 640,
+      show: false,
+      'node-integration': false,
+    });
+
+    // Register simplenote:// protocol
+    protocol.registerHttpProtocol('simplenote', req => {
+      this.authWindow.loadURL(req.url);
+    });
+
+    this.authWindow.webContents.on('will-navigate', (event, url) =>
+      this.onBrowserNavigate(url)
+    );
+
+    this.authWindow.webContents.on(
+      'did-get-redirect-request',
+      (event, oldUrl, newUrl) => this.onBrowserNavigate(newUrl)
+    );
+  };
+
+  onBrowserNavigate = url => {
+    try {
+      const parsedUrl = new URL(url);
+      // Continue on if the url is not the simplenote protocol
+      if (parsedUrl.protocol !== 'simplenote:') {
+        return;
+      }
+
+      this.processUrl(parsedUrl);
+    } catch (error) {
+      return;
+    }
+  };
+
+  processUrl = url => {
```

Comment:
`processUrl` is what it _does_ but not really why it exists. could we think of a more intentional name?

`authenticateFromUrl(url)` for example?

Reply:
taken further we can simplify this `onBrowserNavigate` function if we say "`authenticateFromUrl` exists to handle URL-based authentication reponders…" and then move the try/catch logic in there…

```js
onBrowserNavigate = url => {
  try {
    this.authenticateFromUrl( new Url( url ) );
  } catch ( e ) {
    // failure to authenticate likely means
    // it wasn't an authentication URL
    // so do nothing
  }
}

authenticateFromUrl = url => {
    if ( parsedUrl.protocol !== 'simplenote:' ) {
      return;
    }

    …
}
```